### PR TITLE
Digitalize Super Tank

### PR DESCRIPTION
**Pull request by Quarri6343**
This PR ports [IMPACT](https://github.com/GT-IMPACT/IMPACT-MODPACK) modpack Super Tank feature to GT:NH
4GName has changed its license to [MIT](https://github.com/GT-IMPACT/GT5.09OFFBEAT/blob/master/LICENSE) recently so we can port some contents over without worrying about GT++

Super Tank (and of course, Quantum Tank) now has:

・Auto Output Mode
Like any other machines, toggle "input from output face" with Screwdriver
Output rate depends on its tier

・Fluid Lock Mode
It locks fluid in the slot
You can now disallow other fluid to be inserted into the tank

・OverFlow Voiding Mode
It is same with the old Overflow voiding mode, But Screwdriver are no longer needed

・Full Voiding Mode
If this mode is enabled, the tank will act as a trash can
you can void any fluid without breaking the tank

・GUI Buttons to toggle all these features

https://user-images.githubusercontent.com/39122497/187179567-6e175269-c0e1-4706-a1fc-b34be92046a0.mp4

### CODE REVIEW

**Comment by AlexeyGrishankov** [comment]
good:)

**Review by Dream-Master — APPROVED** [review]

File: src/main/java/gregtech/api/gui/GT_Container.java [hunk 1]
```diff
@@ -670,7 +670,6 @@ protected static ItemStack drainFluid(IFluidAccess aFluidAccess, EntityPlayer aP
             }
             replaceCursorItemStack(aPlayer, tFilledContainer);
         }
-        if (tTankStack.amount <= 0) aFluidAccess.set(null);
```
Comment (Glease) [inline src/main/java/gregtech/api/gui/GT_Container.java:673]: This line should not have been removed. It will leave empty fluid stacks behind.

Comment (Quarri6343) [reply src/main/java/gregtech/api/gui/GT_Container.java:673]: which part went wrong by this removal?
I have to remove this line to lock fluid (leave empty fluid stacks behind) in my tank

Comment (Quarri6343) [reply src/main/java/gregtech/api/gui/GT_Container.java:673]: I see
I have to do something different

Comment (Glease) [reply src/main/java/gregtech/api/gui/GT_Container.java:673]: dream's message

Comment (GlodBlock) [reply src/main/java/gregtech/api/gui/GT_Container.java:673]: the output hatch lock slot has empty fluidstack

**Pull request merged**